### PR TITLE
common/rational general refactoring

### PR DESCRIPTION
**Pull request by lightyear15**
Let's make rational shorter, cleaner and faster.

Quick refactoring of rational class, this commit changes the way
rational numbers behave:
- rational::rational(x) = x / 1
- NaN is indicated by denom == 0
- NaN spreads to everything it touches ( NaN + x = Nan, ...)
- default constructor builds a NaN
- comparison against a NaN behaves like c++ NaN for floating point
numbers :
   * NaN [<,<=,>,>=,==] [x,NaN] == false
   * NaN != [x,NaN] == true

### CODE REVIEW

**Comment by itsmattkc** [comment]
Wow, I'll have to test this out. If it works as well as it looks, this would definitely be much cleaner.

**Comment by lightyear15** [comment]
> 
> 
> Wow, I'll have to test this out. If it works as well as it looks, this would definitely be much cleaner.

Many changes are just some reformatting using the .clang-format.
Very annoying when reviewing, sorry about that

File: app/common/rational.cpp [hunk 1]
```diff
@@ -3,421 +3,312 @@
...
-      numer /= d;
-      denom /= d;
-    }
-
-  if (neg) {
-    numer = -numer;
-  }
+    intType d = gcd(std::abs(numer), denom);
+    numer /= d;
+    denom /= d;
```
Comment (lightyear15) [inline app/common/rational.cpp]: mmm... just reviewing my own changes... this might fail in case ``denom == 0``
this needs an ``if``

**Comment by itsmattkc** [comment]
Sorry it's taken so long to get around to this, development has been somewhat messy for a while but things are finally starting to solidify. Are you still interested in having this merged?

**Comment by lightyear15** [comment]
> Sorry it's taken so long to get around to this, development has been somewhat messy for a while but things are finally starting to solidify. Are you still interested in having this merged?

No particular interest, yet if it interests you, I will be happy to carry it on and make it mergeable

**Pull request closed without merging**